Wikipedia/Code/ViewController.swift

```diff
@@ -331,9 +343,143 @@ class ViewController: PreviewingViewController, NavigationBarHiderDelegate {
         navigationBar.apply(theme: theme)
         applyThemeToCloseButton()
         scrollView?.refreshControl?.tintColor = theme.colors.refreshControlTint
+        
+        toolbar.setBackgroundImage(theme.navigationBarBackgroundImage, forToolbarPosition: .any, barMetrics: .default)
+        toolbar.isTranslucent = false
+        
+        secondToolbar.setBackgroundImage(theme.clearImage, forToolbarPosition: .any, barMetrics: .default)
+        secondToolbar.setShadowImage(theme.clearImage, forToolbarPosition: .any)
+    }
+    
+    // MARK: Toolbars
+    
+    static let toolbarHeight: CGFloat = 44
+    static let constrainedToolbarHeight: CGFloat = 32
+    static let secondToolbarSpacing: CGFloat = 8
+    static let toolbarAnimationDuration: TimeInterval = 0.3
+    
+    var toolbarHeightForCurrentSafeAreaInsets: CGFloat {
+        return view.safeAreaInsets.top == 0 ? ViewController.constrainedToolbarHeight : ViewController.toolbarHeight
+    }
+
+    lazy var toolbar: UIToolbar = {
+        let tb = UIToolbar()
+        tb.translatesAutoresizingMaskIntoConstraints = false
+        return tb
+    }()
+    
+    lazy var toolbarHeightConstraint: NSLayoutConstraint = {
+        return toolbar.heightAnchor.constraint(equalToConstant: toolbarHeightForCurrentSafeAreaInsets)
+    }()
+    
+    lazy var toolbarVisibleConstraint: NSLayoutConstraint = {
+        return view.safeAreaLayoutGuide.bottomAnchor.constraint(equalTo: toolbar.bottomAnchor)
+    }()
+    
+    lazy var toolbarHiddenConstraint: NSLayoutConstraint = {
+        return view.bottomAnchor.constraint(equalTo: toolbar.topAnchor)
+    }()
+    
+    lazy var secondToolbar: UIToolbar = {
+        let tb = UIToolbar()
+        tb.translatesAutoresizingMaskIntoConstraints = false
+        return tb
+    }()
+    
+    lazy var secondToolbarHeightConstraint: NSLayoutConstraint = {
+        return secondToolbar.heightAnchor.constraint(equalToConstant: ViewController.toolbarHeight)
+    }()
+    
+    lazy var secondToolbarVisibleConstraint: NSLayoutConstraint = {
+        return toolbar.topAnchor.constraint(equalTo: secondToolbar.bottomAnchor, constant: ViewController.secondToolbarSpacing)
+    }()
+    
+    lazy var secondToolbarHiddenConstraint: NSLayoutConstraint = {
+        return secondToolbar.topAnchor.constraint(equalTo: toolbar.topAnchor)
+    }()
+    
+    func setupToolbars() {
```

Comment:
Good to see these consolidated to one place - I'm a little wary of every ViewController owning and having two in their hierarchy since this is called on viewDidLoad(). Maybe we can at least have a showsToolbars boolean to bypass for subclasses that don't need it?

Reply:
Good idea - I adjusted it so that VCs that want toolbars need to enable them by calling an `enableToolbars()` func